include/EVT/manager.hpp

```diff
@@ -7,195 +7,260 @@
+*/
+#ifdef RTC_SUPPORTED
 RTC& getRTC() {
-#ifdef STM32F3xx
+    #ifdef STM32F3xx
     static RTCf3xx rtc;
     return rtc;
-#endif
+    #endif
 }
+#endif
 
+#ifdef MCU_SUPPORTED
 template<MCUTimer mcuTimer>
 Timer& getTimer(uint32_t clockPeriod) {
-#ifdef STM32F3xx
+    #ifdef STM32F3xx
     static Timerf3xx timer(getTIM(mcuTimer), clockPeriod);
     return timer;
-#endif
+    #endif
 }
+#endif
 
 }// namespace EVT::core::DEV
 
 namespace EVT::core::IO {
 
 /**
- * Get an instance of an ADC channel
- *
- * @param[in] pin The pin to use with the ADC
- */
+* Get an instance of an ADC channel
+*
+* @param[in] pin The pin to use with the ADC
+*/
+#ifdef ADC_SUPPORTED
 template<Pin pin>
 ADC& getADC() {
-#ifdef STM32F3xx
+    #ifdef STM32F4xx
+    static ADCf4xx adc(pin);
+    return adc;
+    #endif
+    #ifdef STM32F3xx
     static ADCf3xx adc(pin);
     return adc;
-#endif
+    #endif
 }
+#endif
 
 /**
- * Get an instance of a CAN interface.
- *
- * @param[in] txPin CAN pin for transmitting messages
- * @param[in] rxPin CAN pin for receiving messages
- */
+* Get an instance of a CAN interface.
+*
+* @param[in] txPin CAN pin for transmitting messages
+* @param[in] rxPin CAN pin for receiving messages
+*/
+#ifdef CAN_SUPPORTED
 template<Pin txPin, Pin rxPin>
 CAN& getCAN(bool loopbackEnabled = false) {
-#ifdef STM32F3xx
+    #ifdef STM32F3xx
     static CANf3xx can(txPin, rxPin, loopbackEnabled);
     return can;
-#endif
+    #endif
+    #ifdef STM32F4xx
+    static CANf4xx can(txPin, rxPin, loopbackEnabled);
+    return can;
+    #endif
 }
+#endif
 
 /**
- * Get an instance of a GPIO pin.
- *
- * @param[in] pin The pin to attach to the GPIO
- * @param[in] direction The direction, either input or output
- * @param[in] pull The direction of the internal pull resistor
- */
+* Get an instance of a GPIO pin.
+*
+* @param[in] pin The pin to attach to the GPIO
+* @param[in] direction The direction, either input or output
+* @param[in] pull The direction of the internal pull resistor
+*/
+#ifdef GPIO_SUPPORTED
 template<Pin pin>
 GPIO& getGPIO(GPIO::Direction direction = GPIO::Direction::OUTPUT,
               GPIO::Pull pull = GPIO::Pull::PULL_DOWN) {
-#ifdef STM32F3xx
+    #ifdef STM32F3xx
     static GPIOf3xx gpioPin(pin, direction, pull);
     return gpioPin;
-#endif
+    #endif
+    #ifdef STM32F4xx
+    static GPIOf4xx gpioPin(pin, direction);
+    return gpioPin;
+    #endif
 }
+#endif
 
 /**
- * Get an I2C master interface.
- *
- * @param[in] scl The I2C clock pin
- * @param[in] sda The I2C data pin
- */
+* Get an I2C master interface.
+*
+* @param[in] scl The I2C clock pin
+* @param[in] sda The I2C data pin
+*/
+#ifdef I2C_SUPPORTED
 template<Pin scl, Pin sda>
 I2C& getI2C() {
-#ifdef STM32F3xx
+    #ifdef STM32F3xx
     static I2Cf3xx i2c(scl, sda);
     return i2c;
-#endif
+    #endif
 }
+#endif
 
 /**
- * Get an instance of a PWM pin.
- *
- * @param[in] pin The pin to attach to the PWM.
- */
+* Get an instance of a PWM pin.
```

Comment:
Why did you de-dent all these comments? The `*` should be in line with the first `*` on the first line